### PR TITLE
NK1

### PR DESCRIPTION
**Pull request by yiancar**
Just a single big switch macropad. WS2812 is all she has.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — APPROVED** [review]
Too many buttons ;)

**Comment by yiancar** [comment]
one button to rule them all ;)
Pretty basic board :D

File: keyboards/nk1/nk1.h [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "quantum.h"
+
+#define LAYOUT_all( \
```
Comment (fauxpark) [inline keyboards/nk1/nk1.h:21]:
```suggestion
#define LAYOUT_ortho_1x1( \
```

File: keyboards/nk1/info.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+    "keyboard_name": "NK1", 
+    "url": "www.yiancar-designs.com",
+    "maintainer": "yiancar", 
+    "width": 1, 
+    "height": 1, 
+    "layouts": {
+        "LAYOUT_all": {
```
Comment (fauxpark) [inline keyboards/nk1/info.json:8]:
```suggestion
        "LAYOUT_ortho_1x1": {
```

File: keyboards/nk1/rules.mk [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = yes           # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
+AUDIO_ENABLE = no           # Audio output
```
Comment (fauxpark) [inline keyboards/nk1/rules.mk:22]:
```suggestion
AUDIO_ENABLE = no           # Audio output

LAYOUTS = ortho_1x1
```

File: keyboards/nk1/nk1.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    K00 \
+) { \
+    { K00 }, \
```
Comment (fauxpark) [inline keyboards/nk1/nk1.h:24]:
```suggestion
    { K00 } \
```

File: keyboards/nk1/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all( /* Base */
+    KC_ENT),
```
Comment (fauxpark) [inline keyboards/nk1/keymaps/default/keymap.c:20]:
```suggestion
[0] = LAYOUT_ortho_1x1( /* Base */
    KC_ENT)
```

File: keyboards/nk1/keymaps/via/keymap.c [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+[0] = LAYOUT_all( /* Base */
+    KC_ENT),
```
Comment (fauxpark) [inline keyboards/nk1/keymaps/via/keymap.c:20]:
```suggestion
[0] = LAYOUT_ortho_1x1( /* Base */
    KC_ENT)
```

File: keyboards/nk1/keymaps/via/keymap.c [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+};
```
Comment (fauxpark) [inline keyboards/nk1/keymaps/via/keymap.c:21]: Also need three more layers, or set the number of layers to 1 in config.h.

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/nk1/keymaps/via/keymap.c (hunk 7 above)
Comment (yiancar) [reply keyboards/nk1/keymaps/via/keymap.c:21]: Fixed!

**Comment by yiancar** [comment]
Cheers!
I had not checked that `LAYOUT_ortho_1x1` existed

**Review by fauxpark — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**